plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -114,17 +158,15 @@ export const Reader = ({ entityId, onReady }: Props) => {
         baseUrl: window.location.origin,
         onClick: (_: MouseEvent, url: string) => {
           const parsedUrl = new URL(url);
+          if (newerDocsExist && isSynced) {
+            // link navigation will load newer docs
+            setNewerDocsExist(false);
+          }
           if (parsedUrl.hash) {
-            history.pushState(
-              null,
-              '',
-              `${parsedUrl.pathname}${parsedUrl.hash}`,
-            );
+            navigate(`${parsedUrl.pathname}${parsedUrl.hash}`);
           } else {
```

Comment:
Can we keep the `history.pushState` ? Clicking on a hashed link should not trigger a page load. (You can test the correct behavior on the master branch). Relevant PR fixing this problem https://github.com/backstage/backstage/pull/4041

![Screenshot 2021-03-16 at 09 57 42](https://user-images.githubusercontent.com/8065913/111282392-3f1ea780-863e-11eb-9f9f-3700630b196a.png)

Reply:
Currently in master the existing `navigate() `call (just below) causes the doc to update without reloading page.
If I replace it with `history.pushState()` the url in address bar updates but the doc does not update.
It looks like the `if (parsedURl.hash)` part of the if condition isn't being matched so the existing pushState call isn't being used?
Yes, oddly enough this seems to contradict the fix in #4041.

Reply:
@OrkoHunter if you have a moment to look at above

Reply:
> what situation does parsedUrl.hash have a value

When you click on one of those headings as shown in the image, which leads to a URL which is `.../to/some/page/#Heading`.

As of now (in this PR's branch) if you click on one of those hashed heading links on a page, you can see the page being reloaded. However, in the master branch, this behavior is handled by using `history.pushState` implemented in #4041.

I think we need to continue using `history.pushState` or else, we'll introduce the bug once again here.

Reply:
Ok will revert